afids_utils/tests/test_transforms.py

```diff
@@ -1,24 +1,72 @@
 from __future__ import annotations
 
 import numpy as np
+import pytest
 from hypothesis import HealthCheck, given, settings
 from numpy.typing import NDArray
 
-from afids_utils.tests.strategies import affine_xfm, world_coord
-from afids_utils.transforms import afid_world2voxel
+from afids_utils.afids import AfidPosition, AfidVoxel
+from afids_utils.tests.strategies import (
+    affine_xfms,
+    voxel_coords,
+    world_coords,
+)
+from afids_utils.transforms import voxel_to_world, world_to_voxel
 
 
 class TestAfidWorld2Voxel:
-    @given(world_coord=world_coord(), nii_affine=affine_xfm())
+    @given(world_coord=world_coords(), nii_affine=affine_xfms())
     @settings(
         deadline=400,
         suppress_health_check=[HealthCheck.function_scoped_fixture],
     )
-    def test_transform_output(
-        self, world_coord: NDArray[np.float_], nii_affine: NDArray[np.float_]
+    def test_world_to_voxel_xfm(
+        self, world_coord: AfidPosition, nii_affine: NDArray[np.float_]
     ):
-        voxel_coord = afid_world2voxel(world_coord, nii_affine)
+        voxel_coord = world_to_voxel(world_coord, nii_affine)
 
-        assert isinstance(voxel_coord, np.ndarray)
-        assert voxel_coord.dtype == np.int_
-        assert voxel_coord.shape == (3,)
+        assert isinstance(voxel_coord, AfidVoxel)
+        # Have to assert specific int dtype
+        assert isinstance(voxel_coord.i, np.int64)
+        assert isinstance(voxel_coord.j, np.int64)
+        assert isinstance(voxel_coord.k, np.int64)
+
+    @given(voxel_coord=voxel_coords(), nii_affine=affine_xfms())
+    @settings(
+        deadline=400,
+        suppress_health_check=[HealthCheck.function_scoped_fixture],
+    )
+    def test_invalid_world_type(
+        self, voxel_coord: AfidVoxel, nii_affine: NDArray[np.float_]
+    ):
+        with pytest.raises(TypeError, match="Not an AfidPosition.*"):
+            world_to_voxel(voxel_coord, nii_affine)
+
+
+class TestAfidVoxel2World:
```

Comment:
It would be nice to add a round-trip test that confirms that going back and forth yields a value close to the original (I think numpy has some method for "floats approximately equal"

Reply:
Added some tests for this, though really had to loosen the approximation for the `world_to_voxel` round trip tests to pass. I think this is a combination of the rounding errors and also a lack of constraints on the affine matrix.